pyoozie/model.py

```diff
@@ -0,0 +1,580 @@
+from __future__ import unicode_literals
+
+from collections import namedtuple
+from datetime import datetime
+import re
+
+from enum import Enum
+import untangle
+
+from pyoozie.exceptions import OozieException
+
+
+_StatusValue = namedtuple('_StatusValue', ['status_id', 'is_active', 'is_running', 'is_suspendable', 'is_suspended'])
+
+
+def _status(status_id, is_active=False, is_running=False, is_suspendable=False, is_suspended=False):
+    if is_running and not is_active:
+        raise OozieException.parse_error("A running status implies active")
+    return _StatusValue(status_id, is_active, is_running, is_suspendable, is_suspended)
+
+
+_COORD_ID_RE = re.compile('^(?P<id>.*-C)(?:@(?P<action>[1-9][0-9]*))?$')
+_WORKFLOW_ID_RE = re.compile('^(?P<id>.*-W)(?:@(?P<action>.*))?$')
```

Comment:
For consistency I'll rename my regex constants to match these https://github.com/Shopify/pyoozie/issues/7